### ISSUE TITLE
Several issues after deactivating YouTube app

### ISSUE THREAD

**Issue by dagfen** [futo-org/grayjay-android#1553]
### What happened?

I momentarily deactivated my YouTube app on my phone just to check if I could use Grayjay to open YouTube links by default (couldn't do it BTW), but during that process Grayjay* asked for an update and several issues have come up since then.

I was sloppy during several steps though, like sending the logs through the provided button during each instance an exception showed up, but not hitting the share button to paste the relevant log here.

Before issues showed up: I allowed Grayjay to manage Fullscreen in Android App Settings and allowed always opening supported links in App Settings as well.

First Issue: 

Steps to reproduce: Deactivate the YouTube App in App Settings, Open Grayjay and accept to update Grayjay* during normal operation, switch apps for a second during the update operation. Grayjay* will throw an exception after you come back to it.

Problem: Grayjay* wouldn't open or restart even after a few phone restarts, it wouldn't find new updates with the given button*, and would be unusable. Closing Grayjay*  wouldn't fix it, restarting the phone wouldn't fix it, a fresh reinstall wouldn't fix it.

Fix: Reactivating the YouTube App in phone settings.

Second Issue: 

Steps to reproduce: Opening Grayjay again after the first issue showed up, after the first issue was solved, and after Grayjay* was closed for the first time after normal operation, all of this after a reinstall of Grayjay* had been done recently.

Issue: Grayjay* will throw an exception after opening it, restart wouldn't work, checking for updates will show an update, but updating wouldn't work.

Fix: Opening the YouTube App once, then opening Grayjay.

Third Issue: 

Steps to reproduce: Try to update subscriptions feed or open a YouTube video without linking a YouTube account. The frequency of this issue is random.

Problem: The YouTube plugin inside Grayjay asks for captcha, and after you complete it, will keep asking without going away. THIS ISSUE HAS SHOWED UP SEVERAL TIMES BEFORE, it just showed up again during this string of issues.

Fix: It fixes itself randomly after some time or a few restarts.

Fourth Issue:

Steps to reproduce: Set a YouTube video source to 1440p 60 FPS inside Grayjay, after previous issues have happened.

Issue: Video playback will be choppy and the frame rate will drop abysmally at times. Going back through the buffer wouldn't help. Internet bandwidth is enough to support this source with ease.

Fix: Setting resolution to 1080p. Arguably not too much of a problem since 1080p is enough, but I had been using 1440p before this string of issues happened and it worked perfectly.

Device: Xiaomi 12 Lite.
OS: Android 12 SKQ1.211006.001
Player configurations in attached screenshot.

![Screenshot_2024-12-05-14-44-18-155-edit_com futo platformplayer](https://github.com/user-attachments/assets/0b3e6480-7280-410f-9331-bf6d65466956)

### Grayjay Version

268 release

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

217

### When do you experience the issue?

- [ ] While logged in
- [X] While logged out
- [ ] N/A

**Comment by Zvonimir-FUTO** [comment]
This a bit of confusing one, some of your issues are Youtube (app) related and not related to Grayjay and it seems you _fixed_ all of them.

**Comment by dagfen** [comment]
> This a bit of confusing one, some of your issues are Youtube (app) related and not related to Grayjay and it seems you _fixed_ all of them.

Hi! Just doing my due diligence in case those are genuine bugs. My knowledge about software dev is very limited, so excuse me if I reach some wrong conclusions:

Grayjay throwing an exception and me not being able to open the app again (Issue 1 and 2) seems like a very major bug, though. It feels like I should be able to at least use the app with the other plugins I have installed, no?

Issue 3 happens 2-3 times a month and I'm not sure if it's an issue on the YouTube side, or something that is just not registering properly. I'm following the instructions provided by Grayjay (completing the captcha properly, then hitting close) but it keeps popping up non-stop.

Issue 4 is very minor but I wanted to make sure it's not related to the previous problems since it only started happening after that.

Finally I tinkered a lot because I'm a nerd, so I found those fixes, but if they're bugs I don't want them to affect users who might not be as lucky or have the patience for that.

**Comment by Zvonimir-FUTO** [comment]
Well your input is appreciated. Could you just clarify when you mention "Youtube App" did you meant official Youtube android application or the Grayjay plugin?

**Comment by dagfen** [comment]
> Well your input is appreciated. Could you just clarify when you mention "Youtube App" did you meant official Youtube android application or the Grayjay plugin?

The Official Youtube App. Sorry for the confusing wording.

Also to clarify, it was Grayjay that asked for an update, and I'm talking about Grayjay when I'm talking about allowing Fullscreen, allowing to open default links, throwing an exception, sending logs, the app not opening, and many other instances of the word app.

I hadn't noticed the word salad that I made, apologies. (English is not my first language).

**Comment by Zvonimir-FUTO** [comment]
1. First issue has been reported and we made a fix which we are currently testing out for next release. Sorry to hear app went full bonkers on you and won't work at all.

Did you try deleting cache and data before uninstalling the app? 
Did you installed from previous installation apk or did you download latest version?
Which error did you get when you try to install it again?

2. Wasn't able to reproduce second one.

3. There is also issue opened for this one and we are working on a fix for captcha.

4. Couldn't reproduce this one myself. What would you say the reproduction rate is for you for this one? Will check if someone else from the team can reproduce it.

**Comment by dagfen** [comment]
Hey there! No biggie, I've been loving the app despite that temporary issue.

About 1.

> Did you try deleting cache and data before uninstalling the app?

As far as I remember, I didn't.

> Did you installed from previous installation apk or did you download latest version?

I downloaded the latest version back then.

> Which error did you get when you try to install it again?

I don't quite remember, but I think it opened an exception that asked to send logs, and then an error screen that offered to update the app, in that order.

About 2.

It's probably closely linked to the first one since it only happened after that one and was very similar. Hope fixing the first one fixes that one too.

About 3.

Great to hear! It hasn't happened to me for a few months, now that I think about it.

About 4.

This one hadn't happened to me for a long time, but I actually found the root of the issue while trying to reproduce it thanks to your question. 

It only happens **when I set the resolution above 1080p AND the phone's refresh rate is set to 120Hz**. It's also a bit inconsistent, for example I used the video "2020 LG OLED | The Black 4K HDR 60fps" to try and reproduce the issue; the first time I tried with 1440p60 the video lagged, but during subsequent tries I was only able to reproduce the issue with 4K60 sources.

At a glance the reproduction rate is around 5 to 15 FPS.

**Comment by Zvonimir-FUTO** [comment]
Alright that's a nice feedback. Now I can reproduce Issue 4.

Think others might be fixed already. Did you encounter them recently?

**Comment by dagfen** [comment]
No, and I tried to reproduce them and I wasn't able to. They seem fixed!

I'm still not able to set Grayjay as my default app for Youtube links (or any other link for that matter), which is what started all of this, but I'm guessing that warrants opening a different issue.

**Comment by Zvonimir-FUTO** [comment]
No need to open another issue here is one opened for it already https://github.com/futo-org/grayjay-android/issues/1683

**Comment by Zvonimir-FUTO** [comment]
Closing this one as duplicate of #2329 since it's much clearer what needs to be done in that one while this issue is a mess.
